util/queue.py

```diff
@@ -69,6 +69,14 @@ def __init__(self, *args, **kwargs):
         super(Queue, self).__init__(*args, **kwargs, ctx=multiprocessing.get_context())
         self._size = SharedCounter(0)
 
+    # __getstate__ and __setstate__ are needed for pickling, otherwise _size won't be copied.
+    def __getstate__(self):
```

Comment:
Suggestion: instead of concatenating these two tuples, you could instead encapsulate the two pieces of information into an object — for example, a namedtuple with two attributes: 'queue' and 'size'. This approach keeps them separate and makes the code more readable, in particular when we reconstruct the object in `__setstate__()`.